mllib/src/main/scala/org/apache/spark/mllib/feature/PCA.scala

```diff
@@ -44,6 +44,14 @@ class PCA @Since("1.4.0") (@Since("1.4.0") val k: Int) {
     require(k <= numFeatures,
       s"source vector size $numFeatures must be no less than k=$k")
 
+    val workSize = ( 3
+      * math.min(k, numFeatures) * math.min(k, numFeatures)
+      + math.max(math.max(k, numFeatures), 4 * math.min(k, numFeatures)
+      * math.min(k, numFeatures) + 4 * math.min(k, numFeatures))
+      )
+    require(workSize <= Int.MaxValue,
+      "The param K and numFeatures is too large for SVD computation.")
```

Comment:
"K" -> "k"
I'd also add this to be super-explicit: "Try reducing the parameter k for PCA, or reduce the input feature vector dimension to make this tractable."